internal/plugin/resource_plugin_test.go

```diff
@@ -49,37 +50,79 @@ func TestResourceMonitor_Subscriptions(t *testing.T) {
 	assert.Equal(t,
 		[]string{
 			bus.OsProcessesTopic,
+			bus.InstanceConfigContextTopic,
 		},
 		resourcePlugin.Subscriptions())
 }
 
 func TestResource_Instances_Process(t *testing.T) {
 	ctx := context.Background()
-	testResource := protos.GetHostResource()
-
-	fakeResourceService := &servicefakes.FakeResourceServiceInterface{}
-	fakeResourceService.GetResourceReturns(testResource)
-
-	fakeInstanceService := &servicefakes.FakeInstanceServiceInterface{}
-	fakeInstanceService.GetInstancesReturns(testResource.GetInstances())
-
-	resourcePlugin := NewResource(types.GetAgentConfig())
-	resourcePlugin.instanceService = fakeInstanceService
-	resourcePlugin.resourceService = fakeResourceService
-
-	messagePipe := bus.NewFakeMessagePipe()
-	err := messagePipe.Register(2, []bus.Plugin{resourcePlugin})
-	require.NoError(t, err)
 
-	processesMessage := &bus.Message{Topic: bus.OsProcessesTopic, Data: []*model.Process{{Pid: 123, Name: "nginx"}}}
-	messagePipe.Process(ctx, processesMessage)
-	messagePipe.Run(ctx)
-
-	assert.Len(t, messagePipe.GetProcessedMessages(), 2)
-	assert.Equal(t, processesMessage.Topic, messagePipe.GetProcessedMessages()[0].Topic)
-	assert.Equal(t, processesMessage.Data, messagePipe.GetProcessedMessages()[0].Data)
-	assert.Equal(t, bus.ResourceTopic, messagePipe.GetProcessedMessages()[1].Topic)
-	assert.Equal(t, testResource, messagePipe.GetProcessedMessages()[1].Data)
+	nginxConfigContext := &model.NginxConfigContext{
+		AccessLogs: []*model.AccessLog{
+			{
+				Name: "/usr/local/var/log/nginx/access.log",
+			},
+		},
+		ErrorLogs: []*model.ErrorLog{
+			{
+				Name: "/usr/local/var/log/nginx/error.log",
+			},
+		},
+		StubStatus: "http://127.0.0.1:8081/api",
+		InstanceID: protos.GetNginxOssInstance([]string{}).GetInstanceMeta().GetInstanceId(),
+	}
+
+	tests := []struct {
+		name             string
+		processesMessage *bus.Message
+		resource         *v1.Resource
+		topic            string
+	}{
+		{
+			name: "Test 1: OS Process Topic",
+			processesMessage: &bus.Message{
+				Topic: bus.OsProcessesTopic,
+				Data:  map[int32]*model.Process{123: {Pid: 123, Name: "nginx"}},
+			},
+			resource: protos.GetHostResource(),
+			topic:    bus.ResourceTopic,
+		},
+		{
+			name:             "Test 2: Instance Config Context Topic",
+			processesMessage: &bus.Message{Topic: bus.InstanceConfigContextTopic, Data: nginxConfigContext},
+			resource:         protos.GetHostResource(),
+			topic:            bus.ResourceTopic,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
Technically a shadow declaration of `t`.
```suggestion
		t.Run(test.name, func(tt *testing.T) {
```
Everything in the function should then use `tt` instead of `t`.

Reply:
Ditto this in other tests.